From 4f419afe307f808ab648a6d6c473ff3cd0b9d836 Mon Sep 17 00:00:00 2001
From: ivan tkachenko <me@ratijas.tk>
Date: Sun, 10 Jul 2022 15:17:26 +0300
Subject: [PATCH] [kcm/kwindesktop] Emit rowsChanged signal to fix default
 state highlighting

Without this, the rowsChanged signal wasn't emitting at startup, so the
binding `highlight: kcm.desktopsModel.rows !== 2` didn't work correctly.

FIXED-IN: 5.24.7 5.25.3 5.26
---
 src/kcmkwin/kwindesktop/desktopsmodel.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/kcmkwin/kwindesktop/desktopsmodel.cpp b/src/kcmkwin/kwindesktop/desktopsmodel.cpp
index cf147a7981..f8dd8d6260 100644
--- a/src/kcmkwin/kwindesktop/desktopsmodel.cpp
+++ b/src/kcmkwin/kwindesktop/desktopsmodel.cpp
@@ -457,6 +457,8 @@ void DesktopsModel::getAllAndConnect(const QDBusMessage &msg)
         m_names = m_serverSideNames;
         m_rows = m_serverSideRows;
         endResetModel();
+
+        Q_EMIT rowsChanged();
     }
 
     Q_EMIT readyChanged();
-- 
GitLab

